### PR TITLE
[py]: Deprecate `opera` webdriver & options in prep for removal in `4.3`

### PR DESCRIPTION
**Pull request by symonk**
Deprecate opera driver `Options` and `WebDriver` subclass usage.  I have added appropriate 'rip out' todo's so grepping for `# Todo: gh-10379 (Remove for 4.3)` will yield everything that needs removed after 4.2 is cut; and we can properly remove it for `4.3` .  There isn't actually any `opera` webdriver based tests outside of basic unit/options ones so I didn't invest the time into that here as i'll be removing it in a few days again.

relates to #10379

### CODE REVIEW

File: py/selenium/webdriver/__init__.py [hunk 1]
```diff
@@ -25,7 +25,7 @@
 from .edge.webdriver import WebDriver as Edge  # noqa
 from .edge.webdriver import WebDriver as ChromiumEdge  # noqa
 from .edge.options import Options as EdgeOptions  # noqa
-from .opera.webdriver import WebDriver as Opera  # noqa
+from .opera.webdriver import WebDriver as Opera  # noqa  # Todo: gh-10379 (Remove for 4.3)
```
Comment (titusfortner) [inline py/selenium/webdriver/__init__.py]: Are these `Todo:` necessary for user visibility or simply a note for devs? If the latter than let's not have them in there, we'll know what to delete. :)

**Review by titusfortner — CHANGES_REQUESTED** [review]
We probably also should put a link to a note somewhere somehow for why it is getting deprecated, and how to drive Opera with chromedriver. Pretty sure the latter needs to be in our docs with examples.

File: py/selenium/webdriver/__init__.py (hunk 1 above)
Comment (symonk) [reply py/selenium/webdriver/__init__.py]: seems fair; will remove

File: py/selenium/webdriver/opera/options.py [hunk 2]
```diff
@@ -23,7 +24,8 @@ class Options(ChromeOptions):
     KEY = "operaOptions"
 
     def __init__(self):
-        ChromeOptions.__init__(self)
+        warnings.warn(f"{self.__class__} is deprecated and will be removed in 4.3", DeprecationWarning, stacklevel=2)
```
Comment (titusfortner) [inline py/selenium/webdriver/opera/options.py]:
```suggestion
        warnings.warn(f"{self.__class__} is deprecated and will be removed in 4.3; see https://www.selenium.dev/documentation/webdriver/getting_started/open_browser/#opera", DeprecationWarning, stacklevel=2)
```

File: py/selenium/webdriver/opera/webdriver.py [hunk 3]
```diff
@@ -42,6 +44,7 @@ def __init__(self, executable_path=None, port=0,
          - service_log_path - Where to log information from the driver.
            capabilities only, such as "proxy" or "loggingPref".
         """
+        warnings.warn(f"{self.__class__} is deprecated and will be removed in 4.3", DeprecationWarning, stacklevel=2)
```
Comment (titusfortner) [inline py/selenium/webdriver/opera/webdriver.py]:
```suggestion
        warnings.warn(f"{self.__class__} is deprecated and will be removed in 4.3; see https://www.selenium.dev/documentation/webdriver/getting_started/open_browser/#opera", DeprecationWarning, stacklevel=2)
```

**Review by titusfortner — APPROVED** [review]

**Pull request merged**